be/src/olap/in_list_predicate.h

```diff
@@ -338,15 +338,29 @@ class InListPredicateBase : public ColumnPredicate {
                     }
                 }
 
-                if constexpr (is_opposite != (PT == PredicateType::IN_LIST)) {
```

Comment:
Maybe we can just change `is_opposite != (PT == PredicateType::IN_LIST)` to `!is_opposite`.

Or change it to
```cpp
                if (is_opposite ^
                    _operator(_values.find(reinterpret_cast<const T&>(data_array[idx])),
                              _values.end())) {
                    sel[new_size++] = idx;
                }
```